File: JavaScript/Data Structure/Graph/graph.js
class Graph {
    constructor() {
        this.adjList = {};
    }

    addVertex(u) {
        if(!this.adjList[u]) {
            this.adjList[u] = [];
        }
    }

    addEdge(u, v) {
        this.addVertex(u);
        this.adjList[u].push(v);
    }
}

const graph = new Graph();

graph.addEdge("A", "B");
graph.addEdge("A", "C");
graph.addEdge("B", "D");
graph.addEdge("C", "E");
graph.addEdge("D", "F");
graph.addEdge("E", "F");

console.log(graph.adjList);

File: Python/Data Structure/Graph/graph.py
class Graph:
    def __init__(self):
        self.adj_list = {}
    
    def add_vertex(self, u):
        if u not in self.adj_list:
            self.adj_list[u] = []
    
    def add_edge(self, u, v):
        self.add_vertex(u)
        self.adj_list[u].append(v)
    

graph = Graph()

graph.add_edge("A", "B")
graph.add_edge("A", "C")
graph.add_edge("B", "D")
graph.add_edge("C", "E")
graph.add_edge("D", "F")
graph.add_edge("E", "F")


print(graph.adj_list)
